apps/framework-cli/src/project.rs

```diff
@@ -110,6 +110,53 @@ pub struct Project {
     pub supported_old_versions: HashMap<String, String>,
 }
 
+impl<'de> Deserialize<'de> for Project {
+    fn deserialize<D>(deserializer: D) -> Result<Project, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        #[derive(Deserialize)]
+        struct ProjectHelper {
+            language: SupportedLanguages,
+            #[serde(default)]
+            redpanda_config: RedpandaConfig,
+            clickhouse_config: ClickHouseConfig,
+            http_server_config: LocalWebserverConfig,
+            #[serde(default)]
+            git_config: GitConfig,
+            #[serde(default = "HashMap::new")]
+            supported_old_versions: HashMap<String, String>,
+        }
+
+        let helper = ProjectHelper::deserialize(deserializer)?;
+
+        let language_project_config = match helper.language {
+            SupportedLanguages::Typescript => {
+                // Here, you would load or initialize the TypeScript project configuration
+                LanguageProjectConfig::Typescript(TypescriptProject::new(
+                    "default_name".to_string(),
+                ))
+            }
+            SupportedLanguages::Python => {
+                // Here, you would load or initialize the Python project configuration
```

Comment:
This looks like WIP comments. Do you mean to fix it up here or are the values populated elsewhere?

Reply:
No I should remove the comments.